nemo/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -272,26 +274,11 @@ def on_train_end(self, trainer, pl_module):
                     logging.debug(f'Last checkpoint {self.last_model_path} already saved')
                 else:
                     super()._save_last_checkpoint(trainer, monitor_candidates)
+            if self.save_context_on_train_end and not self.always_save_context and is_global_rank_zero():
```

Comment:
I'm confused with the docstring and this code. Docstring says save_context_on_train_end does not depend on always_save_context, but looks like it is?

save_context_on_train_end: Whether to dump the artifacts on_train_end **regardless** of whether ``always_save_context`` is ``True``.
            
`if self.save_context_on_train_end **and** not self.always_save_context`

Reply:
That's just because if `always_save_context` is true, the context is already saved to the final checkpoint. That check avoids unnecessarily saving the context twice.

Reply:
Thanks.